packages/dataviews/src/dataviews-layouts/grid/preview-size-picker.tsx

```diff
@@ -39,31 +45,41 @@ function useViewPortBreakpoint() {
 	return null;
 }
 
-export default function DensityPicker( {
-	density,
-	setDensity,
-}: {
-	density: number;
-	setDensity: React.Dispatch< React.SetStateAction< number > >;
-} ) {
+export function useChangeGridColumnsOnViewportChange() {
 	const viewport = useViewPortBreakpoint();
+	const context = useContext( DataViewsContext );
+	const view = context.view as ViewGrid;
 	useEffect( () => {
-		setDensity( ( _density ) => {
-			if ( ! viewport || ! _density ) {
-				return 0;
-			}
-			const breakValues = viewportBreaks[ viewport ];
-			if ( _density < breakValues.min ) {
-				return breakValues.min;
-			}
-			if ( _density > breakValues.max ) {
-				return breakValues.max;
-			}
-			return _density;
-		} );
-	}, [ setDensity, viewport ] );
+		const gridColumns = view.layout?.gridColumns;
```

Comment:
This effect was changed to a hook and essentially changes conditionally the grid columns on viewport change.

Before it was only effective when the view options dropdown was open, so I'm using it at the main `grid` layout code.

https://github.com/user-attachments/assets/4487bf57-46d1-4702-9f82-c6f845e737cb